docs/src/literate-howto/threaded_assembly.jl

```diff
@@ -1,202 +1,316 @@
+        ## Dynamic scheduler spawning `ntasks` tasks where each task will process a chunk of
+        ## (roughly) equal number of cells (`length(color) ÷ ntasks`).
+        scheduler = OhMyThreads.DynamicScheduler(; ntasks)
+        ## Parallelize the loop over the cells in this color
+        OhMyThreads.@tasks for cellidx in color
+            @set scheduler = scheduler
+            ## Obtain a task local scratch and unpack it
+            @local scratch = ScratchData(dh, K, f, cellvalues_template)
+            (; cell_cache, cellvalues, Ke, fe, assembler) = scratch
+            ## Reinitialize the cell cache and then the cellvalues
+            reinit!(cell_cache, cellidx)
+            reinit!(cellvalues, cell_cache)
+            ## Compute the local contribution of the cell
+            assemble_cell!(Ke, fe, cellvalues, C, b)
+            ## Assemble local contribution
+            assemble!(assembler, celldofs(cell_cache), Ke, fe)
         end
     end
-
     return K, f
 end
-
-# The cell assembly function is written the same way as if it was a single threaded example.
-# The only difference is that we unpack the variables from our `scratch`.
-function assemble_cell!(scratch::ScratchValues, cell::Int, K::SparseMatrixCSC,
-                        grid::Grid, dh::DofHandler, C::SymmetricTensor{4, dim}, b::Vec{dim}) where {dim}
-
-    ## Unpack our stuff from the scratch
-    Ke, fe, cellvalues, facetvalues, global_dofs, ɛ, coordinates, assembler =
-         scratch.Ke, scratch.fe, scratch.cellvalues, scratch.facetvalues,
-         scratch.global_dofs, scratch.ɛ, scratch.coordinates, scratch.assembler
-
-    fill!(Ke, 0)
-    fill!(fe, 0)
-
-    n_basefuncs = getnbasefunctions(cellvalues)
-
-    ## Fill up the coordinates
-    nodeids = grid.cells[cell].nodes
-    for j in 1:length(coordinates)
-        coordinates[j] = grid.nodes[nodeids[j]].x
-    end
-
-    reinit!(cellvalues, coordinates)
-
-    for q_point in 1:getnquadpoints(cellvalues)
-        for i in 1:n_basefuncs
-            ɛ[i] = symmetric(shape_gradient(cellvalues, q_point, i))
-        end
-        dΩ = getdetJdV(cellvalues, q_point)
-        for i in 1:n_basefuncs
-            δu = shape_value(cellvalues, q_point, i)
-            fe[i] += (δu ⋅ b) * dΩ
-            ɛC = ɛ[i] ⊡ C
-            for j in 1:n_basefuncs
-                Ke[i, j] += (ɛC ⊡ ɛ[j]) * dΩ
-            end
-        end
-    end
-
-    celldofs!(global_dofs, dh, cell)
-    assemble!(assembler, global_dofs, Ke, fe)
-end;
-
-function run_assemble()
-    n = 20
-    grid, colors = create_colored_cantilever_grid(Hexahedron, n);
-    ip = Lagrange{RefHexahedron,1}()^3
-    dh = create_dofhandler(grid, ip);
-
-    K = allocate_matrix(dh);
-    C = create_stiffness(Val{3}());
-    ## compilation
-    doassemble(K, colors, grid, dh, C, ip);
-    b = @elapsed @time K, f = doassemble(K, colors, grid, dh, C, ip);
-    return b
+nothing # hide
+
+# !!! details "OhMyThreads functional API: `OhMyThreads.tforeach`"
```

Comment:
The code quoting doesn't seem to render properly in the docs for this.

Reply:
https://github.com/JuliaDocs/Documenter.jl/issues/2583